integr8ly/cloud-resource-operator
# Create standalone VPC when "_network" block is defined

Co-authored-by: laurafitzgerald <lfitzger@example.com>

## Overview

Jira: https://issues.redhat.com/browse/INTLY-8103

## Verification Consuming _network Strat Map

_Note : verification must be completed in a fresh cluster and aws account. No remnants of bundled cluster networking (cluster subnets, subnet groups, security groups) should exist in cluster vpc_

_Note : the seeded resource will not fully install, as the security group is not implemented and will cause the install to fail,  the standalone networking is provisioned before the error, the DeleteNetwork()  function is provisionally implemented, so removing the CR will clean up the standalone network. There are follow on tickets for both security group and DeleteNetwork() implementation_

- Clone this branch
- Run `make cluster/prepare`
- Run `make cluster/seed/managed/postgres`
- Run `make run`
- Make note of the `cidrBlock` value in the `_network` strategy map
- Run `aws ec2 describe-vpcs`
- The `cidrBlock` for the vpc name `RHMI Cloud Resource VPC` it should match the `cidrBlock` in the `_network` strategy map
- Make note of the vpc-id
- Run ` aws ec2 describe-subnets --filters "Name=vpc-id,Values=<<noted vpc-id>>"`   
- Ensure two subnets are returned
- Ensure the `cidrBlock` mask for each subnet is one bit greater then the `cidrBlock` mask in the `_network` strategy map
- Update the `cidrBlock` value in the `_network` strategy map     
- Recheck the `cidrBlock` value in the vpc and subnets, these should not change from the original values        

## Verification Cidr Overlap
_Note we can not have both standalone and bundled vpc cidr blocks overlapping_

- Remove CR from previous verification 
- Ensure standalone vpc, subnets and subnet groups have been cleaned
- Run `make cluster/seed/managed/postgres`
- Run `aws ec2 describe-vpcs`
- Note the `cidrBlock` on the cluster vpc, which can be found by the following tag
```
{
    "Key": "kubernetes.io/cluster/<<cluster id>>",
    "Value": "owned"
}
```
- ensure the `_network` type `cidrBlock` value in the `cloud-resources-aws-strategies` config map overlaps with the `cidrBlock` from the cluster vpc
- Run `make run`
- Ensure the following error :
```
failed to create resource network: standalone vpc creation failed: standalone cidr block 10.0.0.0/26 overlaps with cluster vpc cidr block 10.0.0.0/16, update _network strategy to continue vpc creation
```
